addons/mail/static/src/core/common/thread_model.js

```diff
@@ -767,30 +659,14 @@ export class Thread extends Record {
     }
 
     /**
-     * @param {Object} [options]
-     * @param {boolean} [options.sync] Whether to sync the unread message
-     * state with the server values.
+     * @param {Object} [options] used in overrides
      */
-    markAsRead({ sync } = {}) {
+    markAsRead(options) {
         const newestPersistentMessage = this.newestPersistentOfAllMessage;
         if (!newestPersistentMessage && !this.isLoaded) {
-            this.isLoadedDeferred.then(() => new Promise(setTimeout)).then(() => this.markAsRead());
-        }
-        const alreadyReadBySelf = newestPersistentMessage?.isReadBySelf;
-        if (this.selfMember) {
-            this.selfMember.syncUnread = sync ?? this.selfMember.syncUnread;
-            this.selfMember.seen_message_id = newestPersistentMessage;
-        }
-        if (newestPersistentMessage && this.selfMember && !alreadyReadBySelf) {
-            rpc("/discuss/channel/mark_as_read", {
-                channel_id: this.id,
-                last_message_id: newestPersistentMessage.id,
-                sync,
-            }).catch((e) => {
-                if (e.code !== 404) {
-                    throw e;
-                }
-            });
+            this.isLoadedDeferred
```

Comment:
Since we are touching this line, shouldn't we add an early return in this case? The function will be called again once the thread loads. There is no point on going further?